presto-hive/src/main/java/com/facebook/presto/hive/OrcFileWriter.java

```diff
@@ -146,14 +150,15 @@ public void appendRows(Page dataPage)
         Page page = new Page(dataPage.getPositionCount(), blocks);
         try {
             orcWriter.write(page);
+            rowCount += page.getPositionCount();
         }
         catch (IOException | UncheckedIOException e) {
             throw new PrestoException(HIVE_WRITER_DATA_ERROR, e);
         }
     }
 
     @Override
-    public void commit()
+    public Optional<Page> commit()
```

Comment:
Can we have unittests to make sure the stats match the actual file written? A good place is `AbstractTestHiveFileFormats.createTestFile`. Just commit the file, get stats, and assert the file size is the same as the one recorded.